packages/vite/src/node/server/index.ts

```diff
@@ -645,10 +645,11 @@ export async function _createServer(
     file = normalizePath(file)
     await container.watchChange(file, { event: isUnlink ? 'delete' : 'create' })
 
-    if (publicFiles && config.publicDir && file.startsWith(config.publicDir)) {
-      publicFiles[isUnlink ? 'delete' : 'add'](
-        file.slice(config.publicDir.length),
-      )
+    if (config.publicDir && publicFiles) {
+      const publicDir = normalizePath(config.publicDir)
```

Comment:
Nice! Thanks for the fix 🙏🏼 

Would you extract this as `const normalizedPublicDir = normalizePath(config.publicDir)` out of the function so it is precomputed?

Reply:
Thanks!